natsyncd/natsync.cpp

```diff
@@ -0,0 +1,1047 @@
+                }
+                else
+                {
+                    m_natTwiceTable.del(key);
+                    SWSS_LOG_NOTICE("Twice NAT entry with key %s deleted from APP_DB", key.c_str());
+                    m_natTwiceTable.del(reverseEntryKey);
+                    SWSS_LOG_NOTICE("Twice NAT entry with reverse key %s deleted from APP_DB", reverseEntryKey.c_str());
+                }
+            }
+        }
+    }
+    else if (src_ip_natted || dst_ip_natted)
+    {
+        if (src_ip_natted)
+        {
+            if (addFlag)
+            {
+                FieldValueTuple snat_translated_ip("translated_ip", entry.nat_src_ip.to_string());
+                FieldValueTuple dnat_translated_ip("translated_ip", entry.orig_src_ip.to_string());
+
+                fvVector.push_back(snat_type);
+                fvVector.push_back(dynamic_entry);
+                fvVector.push_back(snat_translated_ip);
+
+                reverseFvVector.push_back(dnat_type);
+                reverseFvVector.push_back(dynamic_entry);
+                reverseFvVector.push_back(dnat_translated_ip);
+            }
+
+            if (src_port_natted)
+            {
+                key             += protostr + entry.orig_src_ip.to_string();
+                reverseEntryKey += protostr + entry.nat_src_ip.to_string();
+
+                /* Case of NAPT entry, where SIP is NAT'ted and the L4 port is NAT'ted. */
+                SWSS_LOG_INFO("SNAPT %s conntrack notification", opStr.c_str());
+
+                string src_l4_port      = std::to_string(entry.orig_src_l4_port);
+                string nat_src_l4_port  = std::to_string(entry.nat_src_l4_port);
+
+                key             += ":" + src_l4_port;
+                reverseEntryKey += ":" + nat_src_l4_port;
+
+                std::vector<FieldValueTuple> values;
+                /* We check for existence of reverse nat entry in the app-db because the same dnat static entry
+                 * would be reported as snat entry from the kernel if a packet that is forwarded in the kernel
+                 * is matched by the iptables rules corresponding to the dnat static entry */
+                if (! m_AppRestartAssist->isWarmStartInProgress())
+                {
+                    if ((entryExists = m_naptCheckTable.get(key, values)))
+                    {
+                        for (auto iter : values)
+                        {
+                            if ((fvField(iter) == "entry_type") && (fvValue(iter) == "static"))
+                            {
+                                /* If a matching Static NAPT entry exists in the APP_DB,
+                                 * it has higher priority than the dynamic napt entry. */
+                                SWSS_LOG_INFO("SNAPT %s: static entry exists, not processing the NAPT notification", opStr.c_str());
+                                if (m_AppRestartAssist->isWarmStartInProgress())
```

Comment:
This condition may not be hit ? on the top we are checking  if (! m_AppRestartAssist->isWarmStartInProgress())

Reply:
Yes, this is dead code, will remove it.